src/diffusers/training_utils.py

```diff
@@ -251,6 +251,32 @@ def state_dict(self) -> dict:
             "collected_params": self.collected_params,
         }
 
+    def store(self, parameters: Iterable[torch.nn.Parameter]) -> None:
+        r"""
+        Args:
+        Save the current parameters for restoring later.
+            parameters: Iterable of `torch.nn.Parameter`; the parameters to be
+                temporarily stored.
+        """
+        parameters = list(parameters)
+        self.collected_params = [param.clone() for param in parameters]
```

Comment:
```suggestion
        self.collected_params = [param.detach().cpu().clone() for param in parameters]
```

Think we should force to store parameters on CPU here

Reply:
This way it doesn't get in the way of blowing up GPU memory